edshipyard.py

```diff
@@ -69,92 +87,130 @@ def class_rating(module):
-                fsd = module	# save for range calculation
+                fsd = module  # save for range calculation
+
                 if mods.get('OutfittingFieldType_FSDOptimalMass'):
                     fsd['optmass'] *= mods['OutfittingFieldType_FSDOptimalMass']['value']
+
                 if mods.get('OutfittingFieldType_MaxFuelPerJump'):
                     fsd['maxfuel'] *= mods['OutfittingFieldType_MaxFuelPerJump']['value']
+
             jumpboost += module.get('jumpboost', 0)
 
             for s in slot_map:
                 if slot.lower().startswith(s):
                     loadout[slot_map[s]].append(cr + name)
                     break
+
             else:
                 if slot.lower().startswith('slot'):
                     loadout[slot[-1]].append(cr + name)
+
                 elif __debug__ and not slot.lower().startswith('planetaryapproachsuite'):
-                    print('EDShipyard: Unknown slot %s' % slot)
+                    print('EDShipyard: Unknown slot {}'.format(slot))
 
         except AssertionError as e:
-            if __debug__: print('EDShipyard: %s' % e)
-            continue	# Silently skip unrecognized modules
-        except:
-            if __debug__: raise
+            if __debug__:
+                print('EDShipyard: {}'.format(e))
+
+            continue  # Silently skip unrecognized modules
+
+        except Exception:
+            if __debug__:
+                raise
 
     # Construct description
     ship = ship_map.get(data['ship']['name'].lower(), data['ship']['name'])
-    string = '[%s]\n' % (data['ship'].get('shipName') and ', '.join([ship, data['ship']['shipName']]) or ship)
-    for slot in ['H', 'L', 'M', 'S', 'U', None, 'BH', 'RB', 'TM', 'FH', 'EC', 'PC', 'SS', 'FS', None, 'MC', None, '9', '8', '7', '6', '5', '4', '3', '2', '1']:
+
+    if data['ship'].get('shipName') is not None:
+        _ships = '{}, {}'.format(ship, data['ship']['shipName'])
+    else:
+        _ships = ship
+
+    string = '[{}]\n'.format(_ships)
+
+    SLOT_TYPES = (
+        'H', 'L', 'M', 'S', 'U', None, 'BH', 'RB', 'TM', 'FH', 'EC', 'PC', 'SS', 'FS', None, 'MC', None, '9', '8',
+        '7', '6', '5', '4', '3', '2', '1'
+    )
+    for slot in SLOT_TYPES:
         if not slot:
             string += '\n'
+
         elif slot in loadout:
             for name in loadout[slot]:
-                string += '%s: %s\n' % (slot, name)
-    string += '---\nCargo : %d T\nFuel  : %d T\n' % (cargo, fuel)
+                string += '{}: {}\n'.format(slot, name)
+
+    string += '---\nCargo : {} T\nFuel  : {} T\n'.format(cargo, fuel)
 
     # Add mass and range
     assert data['ship']['name'].lower() in companion.ship_map, data['ship']['name']
     assert companion.ship_map[data['ship']['name'].lower()] in ships, companion.ship_map[data['ship']['name'].lower()]
+
     try:
+        # TODO: broken link
         # https://github.com/cmmcleod/coriolis/blob/master/app/js/shipyard/module-shipyard.js#L184
```

Comment:
I'm not even sure what information the URL would be conveying.  Just lose this line.